internal/cmd/cli/apply/apply.go

```diff
@@ -95,64 +94,89 @@ func globDirs(baseDir string) (result []string, err error) {
 	return
 }
 
+const bundleCreationMaxConcurrency = 4
+
 // CreateBundles creates bundles from the baseDirs, their names are prefixed with
 // repoName. Depending on opts.Output the bundles are created in the cluster or
 // printed to stdout, ...
-func CreateBundles(ctx context.Context, client client.Client, repoName string, baseDirs []string, opts Options) error {
+func CreateBundles(pctx context.Context, client client.Client, repoName string, baseDirs []string, opts Options) error {
 	if len(baseDirs) == 0 {
 		baseDirs = []string{"."}
 	}
 
-	foundBundle := false
-	gitRepoBundlesMap := make(map[string]bool)
-	for i, baseDir := range baseDirs {
-		matches, err := globDirs(baseDir)
-		if err != nil {
-			return fmt.Errorf("invalid path glob %s: %w", baseDir, err)
-		}
-		for _, baseDir := range matches {
-			if i > 0 && opts.Output != nil {
-				if _, err := opts.Output.Write([]byte("\n---\n")); err != nil {
-					return fmt.Errorf("writing to bundle output: %w", err)
-				}
+	// Using an errgroup to manage concurrency
+	// 1. Goroutines will be launched, honouring the concurrency limit, and eventually block trying to write to `bundlesChan`.
```

Comment:
Does this blocking _need_ to happen? I mean, we could use a buffered channel to prevent writes from blocking until the buffer is full. Since we know how many bundles (at most) we'd like to create (`len(baseDirs)`, if I'm not mistaken), we could dimension that buffer accordingly. Or am I missing something? Is there a concern on memory usage for large bundles?
Anyway, this could be an enhancement idea for a future iteration.